src/parser/shaman/restoration/modules/talents/HighTide.js

```diff
@@ -21,22 +21,65 @@ const bounceReduction = 0.7;
  * Every 40000 mana you spend brings a High Tide, making your next 2 Chain Heals heal for an additional 20% and not reduce with each jump.
  */
 
+ /**
+  * Logs for testing High Tide buff usage
+  * 
+  * Double stack log: https://www.warcraftlogs.com/reports/Qb91XvyqtNjaHRPr#fight=21&type=auras&source=11&pins=0%24Separate%24%23244F4B%24healing%240%240.0.0.Any%24137004809.0.0.Shaman%24true%240.0.0.Any%24false%241064&ability=288675
+  * Buff applied pre-pull: https://www.warcraftlogs.com/reports/R4JncHyajt8VQr9h#fight=48&type=auras&source=14&ability=288675
+
+  */
+
 class HighTide extends Analyzer {
   static dependencies = {
     statTracker: StatTracker,
     critEffectBonus: CritEffectBonus,
   };
   healing = 0;
   chainHealTimestamp = 0;
-
   buffer = [];
 
+  // high tide efficiency trackers
+  usedHighTides = 0;
+  unusedHighTides = 0;
+  currentHighTideBuff = 0;
+
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.HIGH_TIDE_TALENT.id);
 
     this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.CHAIN_HEAL), this.chainHeal);
     this.addEventListener(Events.fightend, this.onFightEnd);
+    
+    // these are for tracking high tide efficiency
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.CHAIN_HEAL), this.onChainHealCast);
+    this.addEventListener(Events.applybuff, this.onHighTideBuff);
+    this.addEventListener(Events.applybuffstack, this.onHighTideBuff); //possible to get a re-application of the buff
+    this.addEventListener(Events.removebuff, this.onHighTideRemoveBuff);
+  }
+
+  // in the event of a High Tide buff, if a buff still exists adds it to unused then resets buff counter to 2
+  onHighTideBuff(event) {
+    if(event.ability.guid === 288675) {
+      this.unusedHighTides += this.currentHighTideBuff; //not sure if this is possible
+      this.currentHighTideBuff = 2;
+    }
+  }
+
+  // in the event of a High Tide remove buff, adds any existing buff to unused and sets buff counter to 0
+  onHighTideRemoveBuff(event) {
+    if(event.ability.guid === 288675) {
+      this.unusedHighTides += this.currentHighTideBuff; //if there were leftovers when buff expires
+      this.currentHighTideBuff = 0;
+    }
+  }
+
+  // on a chain heal cast and buff counter is greater than 0, adds one to used counter 
+  // and reduces buff counter by one.
+  onChainHealCast(event) {
+    if(event.ability.name === "Chain Heal" && this.currentHighTideBuff > 0) {
```

Comment:
```suggestion
    if(this.currentHighTideBuff > 0) {
```
It's already guaranteed to be Chain Heal by how you've set up the event listener.

Reply:
fixed